core/state/statedb_test.go

```diff
@@ -727,3 +741,142 @@ func TestMissingTrieNodes(t *testing.T) {
 		t.Fatalf("expected error, got root :%x", root)
 	}
 }
+
+func TestStateDBAccessList(t *testing.T) {
+	// Some helpers
+	addr := func(a string) common.Address {
+		return common.HexToAddress(a)
+	}
+	slot := func(a string) common.Hash {
+		return common.HexToHash(a)
+	}
+
+	memDb := rawdb.NewMemoryDatabase()
+	db := NewDatabase(memDb)
+	state, _ := New(common.Hash{}, db, nil)
+	state.accessList = NewAccessList()
+
+	verifyAddrs := func(a ...string) {
+		t.Helper()
+		for _, address := range a {
+			if !state.AddrInAccessList(addr(address)) {
+				t.Fatalf("expected %x to be in access list", a)
+			}
+		}
+	}
+	verifySlots := func(a string, slots ...string) {
+		if !state.AddrInAccessList(addr(a)) {
+			t.Fatalf("scope missing address/slots %v", a)
+		}
+		for i, s := range slots {
+			if _, slotPresent := state.SlotInAccessList(addr(a), slot(s)); !slotPresent {
+				t.Fatalf("input %d: scope missing slot %v (address %v)", i, s, a)
+			}
+		}
+	}
+
+	state.AddAddrToAccessList(addr("aa"))             // 1
+	state.AddSlotToAccessList(addr("bb"), slot("01")) // 2,3
+	state.AddSlotToAccessList(addr("bb"), slot("02")) // 4
+	verifyAddrs("aa", "bb")
+	verifySlots("bb", "01", "02")
+
+	// Make a copy
+	stateCopy1 := state.Copy()
+	if exp, got := 4, state.journal.length(); exp != got {
+		t.Fatalf("journal length wrong, expected %d got %d", exp, got)
```

Comment:
Go's stdlib has a general style for mismatch errors messages in tests:

`XYZ mismatch: have [...], want [...]`

Would be nice to adhere to the same style.

Reply:
E.g. `journal length mismatch: have %d, want 4`